sharktank_models/llama3.1/test_llama.py

```diff
@@ -22,9 +22,23 @@ def llama_mlir():
     return str(THIS_DIR / "assets/toy_llama.mlir")
 
 
+@pytest.fixture
+def llama_tp2_mlir():
+    return str(THIS_DIR / "assets/toy_llama_tp2.mlir")
+
+
 @pytest.fixture
 def llama_irpa():
-    return str(THIS_DIR / "assets/toy_llama.irpa")
+    return [str(THIS_DIR / "assets/toy_llama.irpa")]
+
+
+@pytest.fixture
+def llama_tp2_irpa():
+    return [
+        str(THIS_DIR / "assets/toy_llama_tp2.irpa"),
+        str(THIS_DIR / "assets/toy_llama_tp2.rank0.irpa"),
+        str(THIS_DIR / "assets/toy_llama_tp2.rank1.irpa"),
+    ]
```

Comment:
Can you comment a bit on how these sharded .irpa files were generated for the toy model and how they would be similarly generated for the full scale model, e.g. from safetensors or gguf?

Could we reuse the same .irpa file for any number of shards? Do we have tools to split and merge?

If I wanted to regenerate these files, would I run https://github.com/nod-ai/shark-ai/blob/main/sharktank/sharktank/models/llama/toy_llama.py with some `--sharding` option? Or run some tools?

Reply:
I added details on generating the files to the `README.md`. Its probably the easiest way to document where / how these are generated.